share/functions/__fish_anypager.fish

```diff
@@ -0,0 +1,31 @@
+function __fish_anypager --description "Print a pager to use"
+    set -l pager
+    # We prefer $PAGER if we have it
+    set -q PAGER
+    and echo $PAGER | read -at pager
+
+    # or even $MANPAGER if we're allowed to
+    if test "$argv[1]" = "--with-manpager"
+        set -q MANPAGER
+        and echo $MANPAGER | read -at pager
+    end
+
+    # We use them if they *exist*
+    if command -q $pager[1]
+        printf %s\n $pager
+        return 0
+    end
+
+    # Cheesy hardcoded list of pagers.
+    for cmd in bat lv most less more
```

Comment:
I don't think we should prefer any concrete pager over `less` because `less` is the most widely used default. Both `man` and `git` use it as default. A user may have installed `bat` to use it for syntax highlighting.

Reply:
I agree, `bat` appearing as my pager in place of `less` was quite unwelcome. I don't think telling users to set `$PAGER` or `$MANPAGER` **and changing the default pager** is a reasonable alternative to defaulting to the most widely used pager that is present on nearly every system.

Reply:
it's fixed in latest master by b83f3b0e9 (__fish_anypager: prefer less over other pagers, 2023-12-17) but not in a release yet unfortunately

Reply:
Thank you very much!